WireAPI/Sources/WireAPI/APIs/UpdateEventsAPI/Event decoding/Conversation/ConversationCreateEventDecoder.swift

```diff
@@ -34,21 +34,161 @@ struct ConversationCreateEventDecoder {
         )
 
         let timestamp = try container.decode(
-            Date.self,
+            UTCTimeMillis.self,
             forKey: .timestamp
         )
 
-        let conversation = try container.decode(
-            Conversation.self,
+        let payload = try container.decode(
+            Payload.self,
             forKey: .payload
         )
 
         return ConversationCreateEvent(
             conversationID: conversationID,
             senderID: senderID,
-            timestamp: timestamp,
-            conversation: conversation
+            timestamp: timestamp.date,
+            conversation: .init(
+                id: payload.id,
+                qualifiedID: payload.qualifiedID,
+                teamID: payload.teamID,
+                type: payload.type,
+                messageProtocol: payload.messageProtocol,
+                mlsGroupID: payload.mlsGroupID,
+                cipherSuite: payload.cipherSuite,
+                epoch: payload.epoch,
+                epochTimestamp: payload.epochTimestamp?.date,
+                creator: payload.creator,
+                members: payload.members?.toAPIModel(),
+                name: payload.name,
+                messageTimer: payload.messageTimer,
+                readReceiptMode: payload.readReceiptMode,
+                access: payload.access,
+                accessRoles: payload.accessRoles,
+                legacyAccessRole: payload.legacyAccessRole,
+                lastEvent: payload.lastEvent,
+                lastEventTime: payload.lastEventTime?.date
+            )
         )
     }
 
+    private struct Payload: Decodable {
```

Comment:
Previously I was just using the API model directly, but that model uses `Date` instances and not the `UTCTime` and `UTCTimeMillis`.  I wasn't sure what was the best approach:

- have a separate private payload struct for the update event, this is inline with other event decoding, comes at the cost of a fair bit of duplication, but perhaps it's warranted.
- key decoding the `Conversation` API model, but change the instances of `Date` to the corresponding `UTCTime` and `UTCTimeMillis`.

If you have some thoughts please let me know.